frb_codegen/src/source_graph.rs

```diff
@@ -209,34 +210,70 @@ impl Module {
         for item in items.iter() {
             match item {
                 syn::Item::Struct(item_struct) => {
-                    let (ident, mirror) = get_ident(&item_struct.ident, &item_struct.attrs);
-                    let ident_str = ident.to_string();
-                    scope_structs.push(Struct {
-                        ident,
-                        src: item_struct.clone(),
-                        visibility: syn_vis_to_visibility(&item_struct.vis),
-                        path: {
-                            let mut path = self.module_path.clone();
-                            path.push(ident_str);
-                            path
-                        },
-                        mirror,
-                    });
+                    let idents = get_ident(&item_struct.attrs);
+                    if idents.is_empty() {
+                        let ident = item_struct.ident.clone();
+                        let ident_str = ident.to_string();
+                        scope_structs.push(Struct {
+                            ident,
+                            src: item_struct.clone(),
+                            visibility: syn_vis_to_visibility(&item_struct.vis),
+                            path: {
+                                let mut path = self.module_path.clone();
+                                path.push(ident_str);
+                                path
+                            },
+                            mirror: false,
+                        });
+                    } else {
+                        for ident in idents {
+                            let ident_str = ident.to_string();
+                            scope_structs.push(Struct {
+                                ident,
+                                src: item_struct.clone(),
+                                visibility: syn_vis_to_visibility(&item_struct.vis),
+                                path: {
+                                    let mut path = self.module_path.clone();
+                                    path.push(ident_str);
+                                    path
+                                },
+                                mirror: true,
+                            });
+                        }
+                    }
```

Comment:
nit: Is it possible to extract a function, or, e.g. something like `idents = get_ident(&item_struct.attrs).or_else(vec![item_struct.ident.clone()])`? Seems we are duplicating some code currently here